src/Dropzone/assets/src/style.css

```diff
@@ -1,72 +1,121 @@
+  background-color: var(--dropzone-background);
+}
+
+.dropzone-container:has(.dropzone-preview:empty) .dropzone-preview-button {
+  display: none;
+}
+
+.dropzone-container:hover,
+.dropzone-on-drag-enter {
+  background-color: var(--dropzone-background-hover);
+  transition: 0.3s;
 }
 
 .dropzone-input {
-    position: absolute;
-    display: block;
-    top: 0;
-    left: 0;
-    width: 100%;
-    height: 100%;
-    opacity: 0;
-    cursor: pointer;
-    z-index: 1;
+  position: absolute;
+  display: block;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  opacity: 0;
+  cursor: pointer;
+  z-index: 1;
 }
 
 .dropzone-preview {
-    display: flex;
-    align-items: center;
-    max-width: 100%;
+  display: grid;
+  gap: var(--dropzone-spacing);
+  grid-template-columns: repeat(auto-fill, var(--dropzone-image-size));
+  grid-template-rows: auto;
+  place-items: stretch;
+  width: 100%;
+  height: 100%;
+}
+
+.dropzone-preview-file {
+  display: flex;
+  flex-direction: column-reverse;
+  align-items: center;
+  justify-content: start;
+  word-wrap: anywhere;
+}
+
+.dropzone-preview-file:hover {
+  filter: brightness(110%);
 }
 
 .dropzone-preview-image {
-    flex-basis: 0;
-    min-width: 50px;
-    max-width: 50px;
-    height: 50px;
-    margin-right: 10px;
-    background-size: contain;
-    background-position: 50% 50%;
-    background-repeat: no-repeat;
+  margin-bottom: var(--dropzone-spacing);
+  width: var(--dropzone-image-size);
+  aspect-ratio: 1;
+  background-size: cover;
+  background-position: 50% 50%;
+  background-repeat: no-repeat;
+  border-radius: var(--dropzone-radius);
+  box-shadow: 0 0 8px var(--dropzone-background-hover);
 }
 
-.dropzone-preview-filename {
-    word-wrap: anywhere;
+.dropzone-no-preview  {
+  margin-bottom: var(--dropzone-spacing);
+  height: var(--dropzone-image-size);
 }
 
-.dropzone-preview-button {
-    position: absolute;
-    top: 0;
-    right: 0;
-    z-index: 1;
-    border: none;
-    margin: 0;
-    padding: 0;
-    width: auto;
-    overflow: visible;
-    background: transparent;
-    color: inherit;
-    font: inherit;
-    line-height: normal;
-    -webkit-font-smoothing: inherit;
-    -moz-osx-font-smoothing: inherit;
-    -webkit-appearance: none;
+.dropzone-no-preview svg {
+  width: 100%;
 }
 
-.dropzone-preview-button::before {
-    content: '×';
-    padding: 3px 7px;
-    cursor: pointer;
+
+.dropzone-preview-file span {
+  font-weight: 300;
+  font-size: 0.9em;
+}
+
+.dropzone-preview-button {
+  position: absolute;
+  top: var(--dropzone-spacing);
+  right: var(--dropzone-spacing);
+  z-index: 1;
+  border: none;
+  margin: 0;
+  background: none;
+  display: grid;
+  place-items: center;
+  cursor: pointer;
 }
 
 .dropzone-placeholder {
-    flex-grow: 1;
-    text-align: center;
-    color: #999;
+  flex-grow: 1;
+  text-align: center;
+}
+
+.dropzone-on-drag-leave {
+  background-color: var(--dropzone-background);
+  transition: 0.3s;
```

Comment:
If this is the only reason to add the 2 classes, maybe it's not 100% required, wdyt ? :) 

(the same thing can be done with `:has` selectors)